sql/core/src/main/scala/org/apache/spark/sql/execution/python/ExtractPythonUDFs.scala

```diff
@@ -94,36 +95,94 @@ object ExtractPythonUDFFromAggregate extends Rule[LogicalPlan] {
  */
 object ExtractPythonUDFs extends Rule[SparkPlan] with PredicateHelper {
 
-  private def hasPythonUDF(e: Expression): Boolean = {
+  private case class LazyEvalType(var evalType: Int = -1) {
```

Comment:
hmmmmm looks messier then I thought .. previous one looks a bit better to me .. wdyt @BryanCutler ?

Reply:
I'm not too fond of the name `LazyEvalType`, makes it sound like something else.  Maybe `CurrentEvalType`?

Reply:
Yeah the idea of the LazyEvalType is a container object that can be set once. Maybe the name LazyEvalType is confusing. I don't think CurrentEvalType is accurate either because the original idea is that we don't change the value once it's set. Maybe call it `EvalTypeHolder` and add docs to explain?